### PR TITLE
Fabric Bot Configuration

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to add the Fabric Bot rules into the repository.

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]

File: .github/fabricbot.json [hunk 1]
```diff
@@ -0,0 +1,1844 @@
...
+      "eventNames": [
+        "issue_comment"
+      ]
+    }
+  },
+  {
+    "taskType": "scheduled",
+    "capabilityId": "ScheduledSearch",
+    "subCapability": "ScheduledSearch",
+    "version": "1.1",
```
Comment (ahsonkhan) [inline .github/fabricbot.json:198]: What defines these versions? Most of the others seem to be 1.0, but this is 1.1. How come?

File: .github/fabricbot.json [hunk 2]
```diff
@@ -0,0 +1,1844 @@
...
+            1
+          ],
+          "timezoneOffset": -7
+        },
+        {
+          "weekDay": 6,
+          "hours": [
+            1
+          ],
+          "timezoneOffset": -7
```
Comment (ahsonkhan) [inline .github/fabricbot.json:249]: What does -7 mean in this context?

File: .github/fabricbot.json [hunk 3]
```diff
@@ -0,0 +1,1844 @@
...
+        },
+        {
+          "weekDay": 2,
+          "hours": [
+            1,
+            7,
+            13,
+            19
+          ],
```
Comment (ahsonkhan) [inline .github/fabricbot.json:323]: Can you explain these please?

File: .github/fabricbot.json [hunk 4]
```diff
@@ -0,0 +1,1844 @@
...
+        "issues",
+        "project_card"
+      ],
+      "taskName": "[Auto Merge] Reset Approvals on Untrusted Changes",
+      "actions": [
+        {
+          "name": "dismissApprovalPullRequest",
+          "parameters": {
+            "dismissAction": "dismissAll",
+            "comment": "Hi @{issueAuthor}.  We've noticed that new changes have been pushed to this pull request.  Because it is set to automatically merge, we've reset the approvals to allow the opportunity to review the updates."
```
Comment (ahsonkhan) [inline .github/fabricbot.json:1026]: Nice!

File: .github/fabricbot.json [hunk 5]
```diff
@@ -0,0 +1,1844 @@
...
+      "eventType": "issue",
+      "eventNames": [
+        "issue_comment"
+      ],
+      "taskName": "[Resolve Workflow] Unresolve Command Without Permissions",
+      "actions": [
+        {
+          "name": "addReply",
+          "parameters": {
+            "comment": "Hi ${contextualAuthor}, only the original author of the issue can ask that it be unresolved.  Please open a new issue with your scenario and details if you would like to discuss this topic with the team."
```
Comment (ahsonkhan) [inline .github/fabricbot.json:1355]: Who's the contextual author? Is it anyone but the person who opened the issue?

File: .github/fabricbot.json [hunk 6]
```diff
@@ -0,0 +1,1844 @@
...
+            "name": "isOpen",
+            "parameters": {}
+          }
+        ]
+      },
+      "eventType": "pull_request",
+      "eventNames": [
+        "pull_request_review"
+      ],
+      "taskName": "[Stale Pull Requests] Remove Stale PR Designation on on Review",
```
Comment (ahsonkhan) [inline .github/fabricbot.json:1519]:
```suggestion
      "taskName": "[Stale Pull Requests] Remove Stale PR Designation on Review",
```

File: .github/fabricbot.json [hunk 7]
```diff
@@ -0,0 +1,1844 @@
...
+              }
+            ]
+          }
+        ]
+      },
+      "eventType": "pull_request",
+      "eventNames": [
+        "issue_comment"
+      ],
+      "taskName": "[Stale Pull Requests] Remove Stale PR Designation on on Comment",
```
Comment (ahsonkhan) [inline .github/fabricbot.json:1588]:
```suggestion
      "taskName": "[Stale Pull Requests] Remove Stale PR Designation on Comment",
```

File: .github/fabricbot.json [hunk 8]
```diff
@@ -0,0 +1,1844 @@
...
+      "eventNames": [
+        "issues",
+        "project_card"
+      ],
+      "taskName": "[Resolve Workflow] Issue Addressed Label Applied",
+      "actions": [
+        {
+          "name": "addReply",
+          "parameters": {
+            "comment": "Hi @${issueAuthor}.  Thank you for opening this issue and giving us the opportunity to assist.  We believe that this has been addressed.  If you feel that further discussion is needed, please add a comment with the text “`/unresolve`” to remove the “issue-addressed” label and continue the conversation."
```
Comment (ahsonkhan) [inline .github/fabricbot.json:1059]: minor nit: We are not consistent with spacing after the period. In some places, we use 2, others we use 1. Consider changing all to use just 1 spacing, or pick one approach and use that :)

```suggestion
            "comment": "Hi @${issueAuthor}. Thank you for opening this issue and giving us the opportunity to assist. We believe that this has been addressed. If you feel that further discussion is needed, please add a comment with the text “`/unresolve`” to remove the “issue-addressed” label and continue the conversation."
```

**Review by ahsonkhan — APPROVED** [review]

File: .github/fabricbot.json (hunk 1 above)
Comment (jsquire) [reply .github/fabricbot.json:198]: This is an export of the existing set of [Fabric Bot](https://portal.fabricbot.ms/bot/?repo=Azure/azure-sdk-for-cpp) rules for the repository.   We made no changes to the current rule set, and I couldn't tell you explicitly what many of these fields mean.

The portal UI where are rules are currently stored is being deprecated, so we'll need to shift to this config-based approach in order for the existing automation to continue to work.   Edits for the rules will still take place in a UI via import, change, export workflow.

File: .github/fabricbot.json (hunk 2 above)
Comment (jsquire) [reply .github/fabricbot.json:249]: Please see my other response.

File: .github/fabricbot.json (hunk 3 above)
Comment (jsquire) [reply .github/fabricbot.json:323]: Please see my other response.

**Comment by danieljurek** [comment]
@ahsonkhan -- These are the rules for this repository that have been in fabric bot https://portal.fabricbot.ms/bot/?repo=Azure/azure-sdk-for-cpp ... They've been exported because the fabric bot configuration UI is being deprecated.

File: .github/fabricbot.json (hunk 1 above)
Comment (jsquire) [reply .github/fabricbot.json:198]: .... and to answer the obvious question, no they do not yet have an official schema that defines the format.   _(that is on their roadmap)_

**Pull request merged**